common/reconciliation/common/interfaces.go

```diff
@@ -38,25 +38,25 @@ type (
 	// InvariantManager represents a manager of several invariants.
 	// It can be used to run a group of invariant checks or fixes.
 	InvariantManager interface {
-		RunChecks(Execution) ManagerCheckResult
-		RunFixes(Execution) ManagerFixResult
+		RunChecks(interface{}) ManagerCheckResult
```

Comment:
Can we either define these as a union type such that they have a type field and pointer fields where one is specified or do what Liang said above?

Reply:
replied above

Reply:
Okay sounds good